src/sframe/parallel_csv_parser.cpp

```diff
@@ -990,22 +989,6 @@ std::map<std::string, std::shared_ptr<sarray<flexible_type>>> parse_csvs_to_sfra
   
   for (auto p : file_and_status) {
     if (p.second == file_status::REGULAR_FILE) {
-      // throw away empty files
-      try {
-        general_ifstream fin(p.first);
```

Comment:
We do want to keep this try catch.

This was added a long time for the (apparently not uncommon) case where you are reading a collection of CSV files (`tc.SFrame.read_csv('blah/*.csv')`), where each CSV file has a header, AND some files are 0 bytes long. Not skipping the 0 byte files cause the CSV header parse to fail resulting in an unexpected parse failure. 

We could catch the parse failure, but it can be hard to differentiate between a malformed CSV file (in which case we should fail), and this very particular use case (in which case we should just skip and continue). Hence it was easier to just check for the 0 byte case specifically before hand.

Reply:
I'm very surprised to hear this use case isn't uncommon... I thought about it when modifying the code here, and it seemed like a use case we wouldn't necessarily want to support. I'll put back the try/catch and implement my pass-through fix only for collections of all zero-byte files.